proto/def.bzl

```diff
@@ -180,12 +180,59 @@ go_proto_library = rule(
     },
     toolchains = [GO_TOOLCHAIN],
 )
-# go_proto_library is a rule that takes a proto_library (in the proto
-# attribute) and produces a go library for it.
 
-def go_grpc_library(**kwargs):
-    # TODO: Deprecate once gazelle generates just go_proto_library
-    go_proto_library(compilers = [Label("//proto:go_grpc")], **kwargs)
+# Add macro for compatibility mode to ease upgrades.
+# See go_grpc_library() for more information.
+def go_proto_library(name, **kwargs):
+    def is_grpc_compiler(label):
```

Comment:
It looks like this could be a private top-level function instead.

Reply:
Done.